components/camel-chatscript/src/main/java/org/apache/camel/component/chatscript/ChatScriptBot.java

```diff
@@ -0,0 +1,148 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.chatscript;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.Socket;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ChatScriptBot {
+
+    private static final transient Logger LOG = LoggerFactory.getLogger(ChatScriptBot.class);
+    String host;
+    int port = 1024;
+    String message;
+    String botName;
+    String userName;
+    boolean initialized;
+
+    public ChatScriptBot(String iHost, int port, String iBotName, String iUserName) {
+        this.host = iHost;
+        this.port = port;
+        this.botName = iBotName;
+        this.userName = iUserName;
+    }
+
+    public String sendChat(String input) {
+        if (!initialized) {
+            return init(null);
+        }
+        ChatScriptMessage g = new ChatScriptMessage(this.userName, this.botName, input);
+        return doMessage(g.toCSFormat());
+    }
+
+    public String sendChat(ChatScriptMessage input) {
+        if (!initialized) {
+            return init(input);
+        }
+        return doMessage(input.toCSFormat());
+    }
+
+    private String doMessage(ChatScriptMessage msg) {
+        return doMessage(msg.toCSFormat());
+    }
+
+    private String doMessage(String msg) {
+        Socket echoSocket;
+        String resp = "";
+
+        try {
+            echoSocket = new Socket(this.host, this.port);
```

Comment:
I don't know how is ChatScript works. Is it free of auth mechanism?

Reply:
There is no authentication mechanism for ChatScript. Applications which make use of CS must build the authentication.